lib/CMakeLists.txt

```diff
@@ -51,6 +51,12 @@ set(SOURCES
 add_llvm_loadable_module(Svf ${SOURCES})
 add_llvm_Library(LLVMSvf ${SOURCES})
 
+# It seems that the Svf library needs to explictly require the CUDD alloc/init/etc so LLVMCudd
```

Comment:
Alright, this built for me on Ubuntu16.04 LT with the following changes:

`diff --git a/lib/CMakeLists.txt b/lib/CMakeLists.txt
index 3c42075..fe7891d 100644
--- a/lib/CMakeLists.txt
+++ b/lib/CMakeLists.txt
@@ -51,12 +51,18 @@ set(SOURCES
 add_llvm_loadable_module(Svf ${SOURCES})
 add_llvm_Library(LLVMSvf ${SOURCES})
 
-# It seems that the Svf library needs to explictly require the CUDD alloc/init/etc so LLVMCudd
-link_directories( ${CMAKE_BINARY_DIR}/lib/Cudd )
+if ( CMAKE_SYSTEM_NAME MATCHES "Darwin")
+   # It seems that the Svf library needs to explictly require the CUDD alloc/init/etc so LLVMCudd
+   link_directories( ${CMAKE_BINARY_DIR}/lib/Cudd )
+   llvm_map_components_to_libnames(llvm_libs bitwriter core ipo irreader instcombine instrumentation target linker analysis scalaropts support )
+   target_link_libraries(LLVMSvf ${llvm_libs})
+   target_link_libraries(Svf LLVMCudd ${llvm_libs})
+else()
+   llvm_map_components_to_libnames(llvm_libs bitwriter core ipo irreader instcombine instrumentation target linker analysis scalaropts support )
+   target_link_libraries(LLVMSvf ${llvm_libs})
+   target_link_libraries(Svf ${llvm_libs})
+endif()
 
-llvm_map_components_to_libnames(llvm_libs bitwriter core ipo irreader instcombine instrumentation target linker analysis scalaropts support )
-target_link_libraries(LLVMSvf ${llvm_libs})
-target_link_libraries(Svf LLVMCudd ${llvm_libs})
 
 if(DEFINED IN_SOURCE_BUILD)
     add_dependencies(Svf intrinsics_gen)`

I can reissue the pull request if that's easier but this is easy, I'm going to focus on the iterator issue below.